File: Basic Programming/Hackerrank 30 days of code python3 solution/Day_12.py
#                      Name : Arif Khan
#                      Judge: HACKERRANK
#                      University: Primeasia University
#                      problem: Day 12: Inheritance
#                      Difficulty: Easy
#                      Problem Link: https://www.hackerrank.com/challenges/30-inheritance/problem
#

class Person:
	def __init__(self, firstName, lastName, idNumber):
		self.firstName = firstName
		self.lastName = lastName
		self.idNumber = idNumber
	def printPerson(self):
		print("Name:", self.lastName + ",", self.firstName)
		print("ID:", self.idNumber)

class Student(Person):
    def __init__(self, firstName, lastName, idNumber, scores):
        super().__init__(firstName, lastName, idNumber)
        self.scores = scores
        
    def calculate(self):
        avg = sum(self.scores)/len(self.scores)
        grades = 'OEAPDT'
        
        gradelist = [100 >= avg >= 90, 90 > avg >= 80, 80 > avg >= 70 , 70 > avg >= 55 , 55 > avg >= 40, 40 > avg]
        
        for i , m in zip(gradelist,grades):
            if i:
                return m

line = input().split()
firstName = line[0]
lastName = line[1]
idNum = line[2]
numScores = int(input()) # not needed for Python
scores = list( map(int, input().split()) )
s = Student(firstName, lastName, idNum, scores)
s.printPerson()
print("Grade:", s.calculate())

File: Basic Programming/Hackerrank 30 days of code c++ solution/Day_12.cpp
/*                    Name : Arif Khan
                      Judge: HACKERRANK
                      University: Primeasia University
                      problem: Day 12: Inheritance
                      Difficulty: Easy
                      Problem Link: https://www.hackerrank.com/challenges/30-inheritance/problem
*/

#include <iostream>
#include <vector>

using namespace std;


class Person{
	protected:
		string firstName;
		string lastName;
		int id;
	public:
		Person(string firstName, string lastName, int identification){
			this->firstName = firstName;
			this->lastName = lastName;
			this->id = identification;
		}
		void printPerson(){
			cout<< "Name: "<< lastName << ", "<< firstName <<"\nID: "<< id << "\n"; 
		}
	
};

class Student :  public Person{
	private:
		vector<int> testScores;  
	public:
        Student(string firstname, string lastname,int identification,vector<int> inArray)
            :Person(firstname,lastname, identification)
            {
            this-> testScores=inArray;
        }
  		// Write your constructor
    char calculate(){
        int sum = 0,i;
        char ch ;
        for( i = 0; i<testScores.size() ;i++)
             sum += testScores[i];
        sum/=i;
        if(sum < 40)
            ch = 'T';
        else if(sum < 55)
            ch = 'D';
        else if(sum < 70)
            ch = 'P';
        else if(sum < 80)
            ch = 'A';
        else if(sum < 90)
            ch = 'E';
        else
            ch = 'O';
            
            return ch;
    }
};


int main() {
	string firstName;
  	string lastName;
	int id;
  	int numScores;
	cin >> firstName >> lastName >> id >> numScores;
  	vector<int> scores;
  	for(int i = 0; i < numScores; i++){
	  	int tmpScore;
	  	cin >> tmpScore;
		scores.push_back(tmpScore);
	}
	Student* s = new Student(firstName, lastName, id, scores);
	s->printPerson();
	cout << "Grade: " << s->calculate() << "\n";
	return 0;
}